### PR TITLE
Watchlist Automation

### PR DESCRIPTION
**Pull request by elforb**
## Proposed Changes
  - Add support for Watchlists in the Solution Automation Tool
  - Add example Watchlist

### CODE REVIEW

File: Tools/Create-Azure-Sentinel-Solution/createSolution.ps1 [hunk 1]
```diff
@@ -619,6 +621,67 @@ foreach ($inputFile in $(Get-ChildItem $path)) {
...
+                                elements   = @(
+                                    [PSCustomObject]@{
+                                        name    = "watchlists-text";
+                                        type    = "Microsoft.Common.TextBlock";
+                                        options = [PSCustomObject]@{
+                                            text = "Azure Sentinel watchlists enable the collection of data from external data sources for correlation with the events in your Azure Sentinel environment. Once created, you can use watchlists in your search, detection rules, threat hunting, and response playbooks. Watchlists are stored in your Azure Sentinel workspace as name-value pairs and are cached for optimal query performance and low latency. Once deployment is successful, the installed watchlists will be available in the Watchlists blade under 'My Watchlists'.";
+                                            link = [PSCustomObject]@{
+                                                label = "Learn more";
+                                                uri = "https://docs.microsoft.com/azure/sentinel/watchlists";
```
Comment (mpo-sec) [inline Tools/Create-Azure-Sentinel-Solution/createSolution.ps1:644]: Should we do an `aka.ms` link instead?

File: Tools/Create-Azure-Sentinel-Solution/examples/Watchlist_ExampleTemplate.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            "name": "[concat(parameters('workspace'),'/Microsoft.SecurityInsights/',parameters('WatchlistId'))]",
+            "apiVersion": "2021-03-01-preview",
+            "properties": {
+                "description": "test description",
+                "displayName": "test watchlist display name",
+                "source": "CSV",
+                "provider": "Microsoft",
+                "numberOfLinesToSkip": 0,
+                "itemsSearchKey": "Username",
+                "rawContent": "Username,Description\ntestName,testDescription",
```
Comment (mpo-sec) [inline Tools/Create-Azure-Sentinel-Solution/examples/Watchlist_ExampleTemplate.json:33]: Okay so we won't load a CSV file or anything, will expect them to put the encoded string of the CSV here

Comment (elforb) [reply Tools/Create-Azure-Sentinel-Solution/examples/Watchlist_ExampleTemplate.json:33]: Yes, in order to ensure the templates are usable as-is.

File: Tools/Create-Azure-Sentinel-Solution/createSolution.ps1 (hunk 1 above)
Comment (elforb) [reply Tools/Create-Azure-Sentinel-Solution/createSolution.ps1:644]: Good call, will use https://aka.ms/sentinelwatchlists

**Review by mpo-sec — APPROVED** [review]

**Review by v-jayakal — APPROVED** [review]

**Pull request merged**